src/Skeleton_blocker/include/gudhi/Skeleton_blocker_simplifiable_complex.h

```diff
@@ -267,7 +267,7 @@ void Skeleton_blocker_complex<SkeletonBlockerDS>::remove_blocker_include_in_simp
 template<typename SkeletonBlockerDS>
 void Skeleton_blocker_complex<SkeletonBlockerDS>::tip_blockers(Vertex_handle a, Vertex_handle b,
                                                                std::vector<Simplex> & buffer) const {
-  for (auto const & blocker : this->const_blocker_range(a)) {
+  for (auto const blocker : this->const_blocker_range(a)) {
```

Comment:
What was the warning for this one?

Reply:
warning: 
**loop variable 'blocker' is always a copy because the range of type**
'Gudhi::skeleton_blocker::Skeleton_blocker_complex<Gudhi::skeleton_blocker::Skeleton_blocker_simple_traits>::Const_complex_blocker_around_vertex_range'
(aka 'iterator_range<Blocker_iterator_around_vertex_internal<__map_const_iterator<__tree_const_iterator<std::__1::__value_type<Gudhi::skeleton_blocker::Skeleton_blocker_simple_traits::Vertex_handle, Gudhi::skeleton_blocker::Skeleton_blocker_simplex<Gudhi::skeleton_blocker::Skeleton_blocker_simple_traits::Vertex_handle> *>, std::__1::__tree_node<std::__1::__value_type<Gudhi::skeleton_blocker::Skeleton_blocker_simple_traits::Vertex_handle, Gudhi::skeleton_blocker::Skeleton_blocker_simplex<Gudhi::skeleton_blocker::Skeleton_blocker_simple_traits::Vertex_handle> *>, void *> *, long> >, Skeleton_blocker_simplex<Gudhi::skeleton_blocker::Skeleton_blocker_simple_traits::Vertex_handle> *const> >') **does not return a reference** [-Wrange-loop-analysis]

Reply:
Note that in my opinion, clang's warning is bogus. While it is true that the code binds a const reference to a temporary, that's a perfectly normal and idiomatic thing to do. In template code, it could very well be that with different template parameters, the iterator would yield a reference, and `auto const&` or `auto&&` are the best way to handle both cases well at the same time.

Reply:
And indeed the warning has changed since then (https://reviews.llvm.org/D73434) and I don't think `auto const&` can trigger it anymore with just `-Wall` (of course Apple's version of clang lags, so it may still warn 2 years later :disappointed: )

Reply:
In general, I agree with you, but in this particular case, I don't think it will ever return a reference, no? But I have to admit, I am not sure how this kind of `for` loops are really handled, so I could be wrong.

Reply:
First, it could start returning a reference if we modify Blocker_iterator_around_vertex_internal (or pass it different template parameters) so its `dereference` starts returning a reference.
Second, without changing the existing code, nothing prevents a user from adding a specialization of Blocker_iterator_around_vertex_internal that does return a reference. Just because one instantiation of the template has some behavior doesn't mean that all possible specializations have it.

Reply:
Anyway, this change was ok, and I don't really care about the ones in #710 since they are in tests / examples, but if you see more of those warnings, let's stop "fixing" them and update the compiler instead.